x-pack/test/api_integration/apis/ml/trained_models/trained_models_list.ts

```diff
@@ -0,0 +1,96 @@
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { getCommonRequestHeader } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+  const esDeleteAllIndices = getService('esDeleteAllIndices');
+
+  describe('GET trained_models_list', () => {
+    let testModelIds: string[] = [];
+
+    before(async () => {
+      await ml.api.initSavedObjects();
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      testModelIds = await ml.api.createTestTrainedModels('regression', 5, true);
+      await ml.api.createModelAlias('dfa_regression_model_n_0', 'dfa_regression_model_alias');
+      await ml.api.createIngestPipeline('dfa_regression_model_alias');
+
+      // Creating an indices that are tied to modelId: dfa_regression_model_n_1
+      await ml.api.createIndex(`user-index_dfa_regression_model_n_1`, undefined, {
+        index: { default_pipeline: `pipeline_dfa_regression_model_n_1` },
+      });
+    });
+
+    after(async () => {
+      await esDeleteAllIndices('user-index_dfa*');
+
+      // delete created ingest pipelines
+      await Promise.all(
+        ['dfa_regression_model_alias', ...testModelIds].map((modelId) =>
+          ml.api.deleteIngestPipeline(modelId)
+        )
+      );
+      await ml.testResources.cleanMLSavedObjects();
+      await ml.api.cleanMlIndices();
+    });
+
+    it('returns a foramtted list of trained model with stats, associated pipelines and indices', async () => {
+      const { body, status } = await supertest
+        .get(`/internal/ml/trained_models_list`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(getCommonRequestHeader('1'));
+      ml.api.assertResponseStatusCode(200, status, body);
+
+      // Created models + system model + model downloads
+      expect(body.length).to.eql(10);
+
+      const dfaRegressionN0 = body.find((v: any) => v.model_id === 'dfa_regression_model_n_0');
+
+      expect(Object.keys(dfaRegressionN0.pipelines).length).to.eql(2);
+
+      const dfaRegressionN1 = body.find((v: any) => v.model_id === 'dfa_regression_model_n_1');
+      expect(Object.keys(dfaRegressionN1.pipelines).length).to.eql(
+        1,
+        `Expected number of pipelines for dfa_regression_model_n_1 to be ${1} (got ${
+          Object.keys(dfaRegressionN1.pipelines).length
+        })`
+      );
+      expect(dfaRegressionN1.indices.length).to.eql(
+        1,
+        `Expected number of indices for dfa_regression_model_n_1 to be ${1} (got ${
+          dfaRegressionN1.indices.length
+        })`
+      );
+    });
+
+    it('returns models without pipeline in case user does not have required permission', async () => {
```

Comment:
Just checking - is this the expected result when opening the map in the expanded row for a non-DFA model as a viewer user. Is there any way we could catch the error and show e.g. a callout in the tab rather than the error toasts?

<img width="505" alt="Screenshot 2024-12-04 at 14 13 13" src="https://github.com/user-attachments/assets/1574ba22-61c0-4371-8f97-e6da777e9a55">

Reply:
This error comes from another endpoint [`/data_frame/analytics/map/{analyticsId}`](https://github.com/darnautov/kibana/blob/1c41a8415aa3851db2fe842c90d344e7a4c2484a/x-pack/plugins/ml/server/routes/data_frame_analytics.ts#L727). I reckon it should be fixed separately.

Reply:
Agreed - this behavior isn't introduced by this PR and happens on `main`. Can you create a separate issue for it please.